### ISSUE TITLE
Fission up should support IPFS hashes like down does

### ISSUE THREAD

**Issue by bmann** [fission-suite/cli#9]
As an advanced IPFS user, I’d like to add / pin / fork hashes that are already on IPFS by using the fission up command. 

`fission up IPFS_HASH`

Submits the hash for pinning by our service and updates DNS. 

Bonus: `fission up example.fission.name` looks up DNSLINK and retrieves the hash. fission down should do the same.

**Comment by agentofuser** [comment]
Is there any chance the subdomain DNS could be used independently of the pinning service?

I think it's a key missing piece of usability missing from the ipfs onboarding story, and I even ended up registering https://hashlinked.website for this purpose but ended up not having time to implement it myself.

Big kudos for thinking of everything, I love what I'm seeing from Fission!

**Comment by expede** [comment]
> Big kudos for thinking of everything, I love what I'm seeing from Fission!

🙌 Yay, I'm glad!

> Is there any chance the subdomain DNS could be used independently of the pinning service?

Depends on what your goals are. There's a flag to only update DNS without the pinning service (`fission up --dns-only`), but not to plug in an arbitrary DNS service.

**Comment by bmann** [comment]
@agentofuser we have a request for supporting custom DNS https://github.com/fission-suite/cli/issues/26

Oh and you CAN use the API directly to update the DNS https://hostless.dev/docs 

Please write more about what you’re trying to accomplish — feel free to file a new issue.

**Comment by agentofuser** [comment]
@expede  Oh, `--dns-only` seems to be what I was looking for, thank you!

@bmann what I'm trying to do is, for example, pin something to Pinata with `ipfs-deploy`, then use `fission up --dns-only` to set the hash as the `dnslink` for `agentofuser.fission.name _without_ pinning it to fission itself. That way I'm only using the subdomain dnslink feature without using up my pinning storage quota.

It seems to have worked: I pinned `QmWowBj2hDNjqzNYLuwSra26rMrbhC4w9WYYtq16Gc9qbR` to Pinata then ran `fission --dns-only public/` and, after a while, I got it to load from https://agentofuser.fission.name.

I then deleted the pinned hash from Pinata, but I'm still seeing it on https://agentofuser.fission.name. Probably due to caching though, right? Could you confirm that the way I did this there is no pinning or quota usage on the Fission side?

It would be nice to have the `IPFS_HASH` parameter though, as right now both `ipfs-deploy` and `fission up` are hashing the contents of `public/`, and if anything changes it in the meantime there would be a mismatch (besides the redundant CPU usage).

Also, if `fission up` printed the hash it just set as the dnslink I'd be more sure that it got the right one.

**Comment by agentofuser** [comment]
@bmann love the hostless.dev domain :)

I tried running the `PUT /dns/{cid}` command from the Swagger API but got an error due to malformed URL. This is the `curl` command that the Web UI generated:

```
curl -X PUT "https://"https://runfission.com"/dns/QmaYSP6h3AvqKCrsS9hvV6Zp7ZoKQTchJ737ezXxcNJj31" -H  "accept: text/plain;charset=utf-8"
```

At the top of the page it says `[ Base URL: "https://runfission.com" ]`, so that's likely the issue.

I fixed the URL and ran it from the command-line and it worked \o/

**Comment by bmann** [comment]
Great feedback. No, there would be no quota usage with just the DNS service. I can confirm that you have no CIDs associated with your user account currently.

Deleting the pin on Pinata doesn't affect the presence or absence of the CID in the IPFS network. If our gateway / node can reach the CID, it will serve it up. So very dependent on when CIDs get garbage collected by different nodes in the network, and whether our nodes have a path to them.

We also know that there is a _mismatch_ between the ipfs-deploy hash and the fission up hash. I need to document this as a known issue -- it's related to how we use the "native" IPFS functionality of a local node, and ipfs-deploy uses web requests, where each file has different metadata.

**Comment by bmann** [comment]
> @bmann love the hostless.dev domain :)

Yeah, we're still debating where/how we're going to use that brand :) We're being consistent with variants of Fission for the moment. runfission wasn't loading for me but hostless was. From your malformed URL we'll need to look into that.

> I fixed the URL and ran it from the command-line and it worked \o/

Yay!

**Comment by agentofuser** [comment]
>I can confirm that you have no CIDs associated with your user account currently.

This is awesome! Thank you for looking it up.

>We also know that there is a mismatch between the ipfs-deploy hash and the fission up hash. I need to document this as a known issue -- it's related to how we use the "native" IPFS functionality of a local node, and ipfs-deploy uses web requests, where each file has different metadata.

Oh, well, in the example I tried both `ipfs-deploy` and `fission up --dns-only` reached the same hash and things worked perfectly. It was a gatsby `public/` directory with lots of stuff. In what scenarios have you observed them diverge?
